tasks/indexer/chainhistoryindexer.go

```diff
@@ -0,0 +1,151 @@
+package indexer
+
+import (
+	"container/list"
+	"context"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	pg "github.com/go-pg/pg/v10"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/label"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/sentinel-visor/lens"
+	"github.com/filecoin-project/sentinel-visor/storage"
+)
+
+func NewChainHistoryIndexer(d *storage.Database, node lens.API) *ChainHistoryIndexer {
+	return &ChainHistoryIndexer{
+		node:    node,
+		storage: d,
+
+		// TODO base finality value on the spec: https://github.com/filecoin-project/specs-actors/pull/702
+		finality: 1400,
```

Comment:
Latest specs-actors has finality at `900` and doesn't appear to be changing soon.

Reply:
Adjusted